src/Components/Web/src/HtmlRendering/HtmlRendererCore.cs

```diff
@@ -7,19 +7,29 @@
 using Microsoft.AspNetCore.Components.Web;
 using Microsoft.Extensions.Logging;
 
-namespace Microsoft.AspNetCore.Components.HtmlRendering;
+namespace Microsoft.AspNetCore.Components.HtmlRendering.Infrastructure;
 
-internal class HtmlRendererCore : Renderer, IHtmlRendererContentProvider
+/// <summary>
+/// A <see cref="Renderer"/> subclass that is intended for static HTML renderering.
+/// </summary>
+public class HtmlRendererCore : Renderer
```

Comment:
* Do you think we could find a different name that does not involve the "Core" suffix?
  * I do not feel super strongly about it, but maybe something like StaticHtmlRenderer or StaticRenderer?

Reply:
OK, I don't feel strongly about this either and have renamed it to `StaticHtmlRenderer` as suggested.

There's a slightly quirky aspect that `HtmlRenderer` wraps this and hence you might think of it as being more specialised, but the naming implies the opposite (as if `StaticHtmlRenderer` was more specialised than `HtmlRenderer`). That's basically the reason for the `...Core` naming originally.

However since we don't expect people to use `StaticHtmlRenderer` directly I don't think matters much, so I'm going with your suggestion as-is.